experimental/packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -245,37 +248,33 @@ export class NodeSDK {
         ? this._resource
         : this._resource.merge(
             new Resource({
-              [SEMRESATTRS_SERVICE_NAME]: this._serviceName,
+              [ATTR_SERVICE_NAME]: this._serviceName,
             })
           );
 
-    // if there is a tracerProviderConfig (traceExporter/spanProcessor was set manually) or the traceExporter is set manually, use NodeTracerProvider
-    const Provider = this._tracerProviderConfig
-      ? NodeTracerProvider
-      : TracerProviderWithEnvExporters;
+    const spanProcessors = this._tracerProviderConfig
+      ? this._tracerProviderConfig.spanProcessors
+      : getSpanProcessorsFromEnv();
 
-    // If the Provider is configured with Env Exporters, we need to check if the SDK had any manual configurations and set them here
-    const tracerProvider = new Provider({
+    this._tracerProvider = new NodeTracerProvider({
       ...this._configuration,
       resource: this._resource,
+      spanProcessors,
     });
 
-    this._tracerProvider = tracerProvider;
-
-    if (this._tracerProviderConfig) {
-      for (const spanProcessor of this._tracerProviderConfig.spanProcessors) {
-        tracerProvider.addSpanProcessor(spanProcessor);
-      }
+    // TODO: the former class `TracerProviderWithEnvExporters` was doing registration only if
+    // spanProcessors were configured. Should we have this logic here?
+    // if we register anyway the context manager changes and break some tests
```

Comment:
I *think* we should not register if there's no `SpanProcessor`- having it there and working through spans instead of it being a no-op introduces additional overhead for the user - so your current implementation is fine. 

What we should consider, though is registering a `ContextManager` and `Propagator` (not in this PR, but later) if no `SpanProcessor` is registered but either a logs SDK or metrics SDK is. Having a `ContextManager` and `Propagator` can be useful in these cases (having a `TraceID` for logs when there's an active trace from another service but no trace SDK in this service can be very helpful and is something that users have asked for previously)

Reply:
Okay, I'll create an issue for that